miio/integrations/vacuum/roborock/vacuum.py

```diff
@@ -866,12 +867,14 @@ def set_dust_collection_mode(self, mode: DustCollectionMode) -> bool:
         return self.send("set_dust_collection_mode", {"mode": mode.value})[0] == "ok"
 
     @command()
+    @action(name="Start dust collection", icon="mdi:turbine")
     def start_dust_collection(self):
         """Activate automatic dust collection."""
         self._verify_auto_empty_support()
         return self.send("app_start_collect_dust")
 
     @command()
+    @action(name="Stop dust collection", icon="mdi:turbine")
```

Comment:
Are you planning to add action decorators for the rest of the actions in a separate PR? It would be fine for `find()` and other actions that do not require any inputs, for those that require input parameters we need to figure out a way to deliver that information to homeassistant.

Reply:
HomeAssistant buttons can only do "press" action, so it is not possible to supply any inputs.
Functions that require inputs need to be implemented diffrently (select, input_number, switch etc.).

Reply:
How about `find()`, `start()`, `stop()`, `pause()`, `home()`, `test_sound_volume()`, the actions for segment & zoned cleaning?

Basically, all actions that can be useful to be called from homeassistant should be marked as such. The homeassistant integration can then ignore creating separate "standard" actions if they are implemented by a specific platform, like vacuum.

Reply:
Are we not making our lives difficult by first marking them as actions that need to be added and then in HomeAssistant hardcoding to ignore those actions (because they are already implemented in the base vacuum entity)?

Reply:
> find() is already included in the vacuum.roborock entitiy, so I don't think we need to add a seperate button.

The idea behind `VacuumInterface` was not probably a suitable one and I have been thinking that it would probably make more sense to codify this using the descriptors. This will make it simpler to support all integrations (and genericmiot devices), as there is no need to inherit from a specific base class (or implement specific methods) to let homeassistant know about the device type. It might be that we will have `genericvacuum` or something similar later on in the future, but I'd concentrate now on getting it done inside homeassistant using the descriptors.

So, at the end, the vacuum platform for homeassistant should look up if there exists an action with `vacuum:start`, `vacuum:pause` or `vacuum:find` (or whatever naming scheme is chosen) and use that to inform both 1) whether the feature is supported, and 2) provide an interface to execute the wanted action.

Reply:
Alright I think this is a good idea, but maybe we schould make a seperate decorator for this like "entity_action" (need to think about a good name).
Because otherwise we will end up inside the button_ng.py file of HomeAssistant with a whole list of exceptions that schould not be added (to prevent doubles).

If we make a seperate decorator for actions that schould not be added as a separate button but be part of a genric entity (like the vacuum entity or fan entity or media_player entity), I think we will make our live easier.

Reply:
The problem is that we still need to handle it somehow based on the IDs (or similar information), as the genericmiot actions are created dynamically from JSON files. One potential way could be keeping a state of the "initialized" descriptors and initialize the "main" platforms first to let them handle the actions they want, leaving the rest uninitialized ones to the "common" platforms to handle.

Reply:
Normally there is no order in the initialization of the diffrent platforms inside HomeAssistant.
For now I will add the action with a extra key "type": "vacuum", that can be used to seperate them in HomeAssistant. We can figure out a better or more generic way in a future PR, but that is out of scope of this PR.

Reply:
Okay, I thought we could have controlled the initialization as the platforms are normally initialized using `async_forward_entry_setups()`. But let's merge this now and take care of that later.